<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.17"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>My Project: Sequencing_Metrics_Tools</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">My Project
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.17 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">Sequencing_Metrics_Tools </div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p>This repo hosts scripts/tools that are used for sequencing metrics calculation. Please refer to this confluence pages for the list of metrics</p>
<p><a href="https://rochewiki.roche.com/confluence/display/RSS/Sequencing+Metrics+Summarization">https://rochewiki.roche.com/confluence/display/RSS/Sequencing+Metrics+Summarization</a></p>
<h1><a class="anchor" id="autotoc_md1"></a>
Install dependencies and pytests</h1>
<p>For metrics, To install dependencies and running scripts / pytest from Github workflows as shown below as example: </p><div class="fragment"><div class="line">python3.10 -m pip install --upgrade pip</div>
<div class="line">python3.10 -m pip install poetry</div>
<div class="line">cd bfx-ngs/metrics</div>
<div class="line">poetry install --no-root</div>
</div><!-- fragment --><div class="fragment"><div class="line"># Run module scripts</div>
<div class="line">poetry run python src/report/summarize_metrics.py [arguments ...]</div>
</div><!-- fragment --><div class="fragment"><div class="line"># Run tests</div>
<div class="line">poetry run pytest</div>
</div><!-- fragment --><p>To run mosdepth - fast BAM/CRAM depth calculation </p><div class="fragment"><div class="line"># Download mosdepth</div>
<div class="line">wget https://github.com/brentp/mosdepth/releases/download/v0.3.4/mosdepth</div>
<div class="line">chmod +x mosdepth</div>
<div class="line"> </div>
<div class="line"># Run mosdepth</div>
<div class="line"># ex.</div>
<div class="line"># ./mosdepth -t 1 --by &#39;bed_file&#39; -Q 4 &#39;sample_name&#39; &#39;sample.bam&#39;</div>
<div class="line">./mosdepth -t 1 --by &#39;GRCh38_major.bed&#39; -Q 4 &#39;NA12878&#39; &#39;NA12878.bam&#39;</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md2"></a>
Protocol Buffer</h2>
<p>To compile proto schema into python file using protoc compiler install protoc compiler and run </p>
<h1><a class="anchor" id="autotoc_md3"></a>
protoc -I=$SRC_DIR â€“python_out=$DST_DIR $SRC_DIR/schema.proto</h1>
<p>This generates schema_pb2.py</p>
<p>for example to generate demux_schema_pb2.py protoc -I=src/demux_stats/ &ndash;python_out=src/demux_stats src/demux_stats/demux_schema.proto</p>
<p>more details can be found <a href="https://protobuf.dev/getting-started/pythontutorial/">here</a></p>
<h1><a class="anchor" id="autotoc_md4"></a>
Docker instructions</h1>
<p>Releases can be found for every tag at: </p><pre class="fragment">navifyplatform-eug-docker.jfrog.io/bfx-sbx-metrics:latest
navifyplatform-eug-docker.jfrog.io/bfx-sbx-metrics:&lt;version&gt;
navifyplatform-eug-docker.jfrog.io/bfx-sbx-metrics:&lt;version&gt;-&lt;branch&gt;-&lt;build&gt;-&lt;commit sha&gt;
</pre><p>Specific versions are available here: <a href="https://navifyplatform.jfrog.io/ui/repos/tree/General/eug-docker/bfx-sbx-qc_metrics?projectKey=eug">https://navifyplatform.jfrog.io/ui/repos/tree/General/eug-docker/bfx-sbx-qc_metrics?projectKey=eug</a></p>
<p>Snapshot can be found at <a href="https://navifyplatform.jfrog.io/ui/repos/tree/General/eug-snap-docker/bfx-sbx-metrics/0.1.0-SNAPSHOT-main-latest?projectKey=eug-snap">https://navifyplatform.jfrog.io/ui/repos/tree/General/eug-snap-docker/bfx-sbx-metrics/0.1.0-SNAPSHOT-main-latest?projectKey=eug-snap</a> </p><pre class="fragment">navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.1.0-SNAPSHOT-main-latest
navifyplatform-eug-docker.jfrog.io/bfx-sbx-metrics:&lt;version&gt;-&lt;branch&gt;-&lt;build&gt;-&lt;commit sha&gt;
navifyplatform-eug-docker.jfrog.io/bfx-sbx-metrics:&lt;version&gt;-&lt;branch&gt;-latest
</pre><h2><a class="anchor" id="autotoc_md5"></a>
Running</h2>
<p>There as an <a href="scripts/test_run_commands_in_docker.sh">example script</a> with all docker commands. These are also in detail below.</p>
<h2><a class="anchor" id="autotoc_md6"></a>
Specific examples</h2>
<h3><a class="anchor" id="autotoc_md7"></a>
Clustering Strand Bias</h3>
<pre class="fragment">run command: python cluster_strand_balance.py --input_file &lt;Clsuter txt file&gt; --output_dir &lt;Output directory&gt; --samplename &lt;Sample Name&gt;
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest cluster_strand_balance.py --input_file `pwd`/metrics/cfDNA1.run_fastq_1m.umis.demux.fq.txt --output_dir `pwd`/qc_metrics/cluster_strand_bias/cfDNA1 --samplename cfDNA1
</pre><h3><a class="anchor" id="autotoc_md8"></a>
Cluster Size Analysis</h3>
<pre class="fragment">run command: python calc_clustersize.py --input_files &lt;Cluster text file&gt; --samplenames &lt;Sample Name&gt; --filetypes AA --output_dir &lt;Output directory&gt;
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest calc_clustersize.py --input_file `pwd`/metrics/cfDNA6.run_fastq_1m.umis.demux.fq.txt --samplenames cfDNA6 --filetype AA --output_dir &lt;Output directory&gt;
</pre><h3><a class="anchor" id="autotoc_md9"></a>
UMI distance</h3>
<pre class="fragment">run command: python umi_distances.py --input_file &lt;Cluster text file&gt;
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest umi_distances.py --input_file `pwd`/metrics/cfDNA1.run_fastq_1m.umis.demux.fq.txt --samplename cfDNA1
</pre><h3><a class="anchor" id="autotoc_md10"></a>
Base Level Accuracy</h3>
<pre class="fragment">run command: python screen_bam_for_errors.py --bam &lt;Consensus BAM file&gt; &gt; error.summary
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest screen_bam_for_errors.py --bam `pwd`/metrics/cfDNA1.run_fastq_1m.umis.demux.fq.bam &gt; error.summary
</pre><h3><a class="anchor" id="autotoc_md11"></a>
Bias in sequences</h3>
<pre class="fragment">run command: python bias_in_sequences.py --input_bam &lt;Consensus BAM file&gt; --samplename &lt;Sample name&gt; --mapq 10 --max_frag_len 2700 --full_gc_range --vmin_vmax_range 0 75 11 33 0 1
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest bias_in_sequences.py --input_bam `pwd`/metrics/cfDNA1.run_fastq_1m.umis.demux.fq.bam --samplename cfDNA1 --mapq 10 --max_frag_len 2700 --full_gc_range --vmin_vmax_range 0 75 11 33 0 1
</pre><h3><a class="anchor" id="autotoc_md12"></a>
Median Raw Read accuracy</h3>
<pre class="fragment">run command: python get_median_rr_acc.py --bam_fname &lt;Raw BAM file&gt; --output_file &lt;Output text file path&gt;
docker command: docker run --rm -v `pwd`:`pwd` navifyplatform-eug-snap-docker.jfrog.io/bfx-sbx-metrics:0.0.8-SNAPSHOT-avc-551-poerty-dockerfile-latest get_median_rr_acc.py --bam_fname `pwd`/metrics/cfDNA1.run_fastq_1m.umis.demux.fq_alignment.bam --output_file `pwd`/cfDNA1.rr.accuracy.txt
</pre><h2><a class="anchor" id="autotoc_md13"></a>
To Contribute</h2>
<p>Please create branch for your development work. Some small testing data can be found here <a href="https://rds-csi.atlassian.net/browse/ATL-401">https://rds-csi.atlassian.net/browse/ATL-401</a>. Please create PR after successful testing and add reviewers (people who are most familar with the topic, or Admin tengl6) to your PR. Admins will merge upon review approval.</p>
<h2><a class="anchor" id="autotoc_md14"></a>
Migration</h2>
<pre class="fragment">git remote add old-repo https://ghe-rss.roche.com/NGS-Algorithms-Applications/Sequencing_Metrics_Tools
git remote set-url origin https://github.com/Roche-DIA-RDS-CSI/Sequencing_Metrics_Tools
</pre><p>Confirm: </p><pre class="fragment">git remote -v
    old-repo    https://ghe-rss.roche.com/NGS-Algorithms-Applications/Sequencing_Metrics_Tools (fetch)
    old-repo    https://ghe-rss.roche.com/NGS-Algorithms-Applications/Sequencing_Metrics_Tools (push)
    origin  https://github.com/Roche-DIA-RDS-CSI/Sequencing_Metrics_Tools (fetch)
    origin  https://github.com/Roche-DIA-RDS-CSI/Sequencing_Metrics_Tools (push)
</pre><p>Push and pull in the standard way. </p>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.17
</small></address>
</body>
</html>
